--- ./jquery.history.js.orig	2009-10-13 17:24:28.000000000 +0200
+++ ./jquery.history.js	2010-11-23 13:49:34.000000000 +0100
@@ -123,7 +123,7 @@
 		var newhash;
 		hash = decodeURIComponent(hash.replace(/\?.*$/, ''));
 		
-		if (jQuery.browser.safari) {
+		if (jQuery.browser.safari || BrowserDetect.browser == 'Konqueror') {
 			newhash = hash;
 		}
 		else {
@@ -156,6 +156,10 @@
 			//      URL in the browser and the "history" object are both updated correctly.
 			location.hash = newhash;
 		}
+		else if (BrowserDetect.browser == 'Konqueror') {
+			location.hash = newhash;
+			jQuery.historyCallback(hash);
+		}
 		else {
 		  jQuery.historyCallback(hash);
 		}
